src/core/client_channel/client_channel.h

```diff
@@ -39,8 +39,17 @@ namespace grpc_core {
 
 class ClientChannel : public Channel {
  public:
-  using PickerObservable =
-      Observable<RefCountedPtr<LoadBalancingPolicy::SubchannelPicker>>;
+  struct NoPicker {
```

Comment:
Changed to the picker approach.

The story here is that the object carried down this far is not the call object but the call spine - so the call object cancels the spine and then unrefs itself and goes away, leaving a cancelled call spine dangling momentarily. It'll be cleaned up because of the `SpawnGuardedUntilCallCompletes` stanza in load balanced call destination, but we might race and see the channel closed first.

Reply:
Oh, interesting.  I somehow had the mental model that promise cancellation was always synchronous, but I guess that can't synchronously handle cleanup of things running in other threads.

Thanks for the explanation!